addons/web/static/tests/views/form_tests.js

```diff
@@ -2254,6 +2255,66 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test("modal should not use the widget footer elements", async function (assert) {
+        assert.expect(2);
+
+        var MyWidget = basicFields.FieldChar.extend({
+            className: 'oe_form_field d-flex',
+            _renderEdit: function () {
+                var res = this._super();
+                const $el = $(
+                    `<section class="oe_form_field">
+                        <header>title</header>
+                        <article>content ` + this.$el[0].outerHTML + `</article>
+                        <footer><button>button</button></footer>
+                    </section>`);
+                this.$el.replaceWith($el);
+                this.$el = $el;
+                return res;
+            },
+        });
+        fieldRegistry.add('test', MyWidget);
+
+        this.data.partner.records[0].reference = 'partner,2';
+
+        var form = await createView({
+            View: FormView,
+            model: 'user',
+            data: this.data,
+            arch: `<form>
+                        <field name="name"/>
+                        <field name="partner_ids">
+                            <tree editable="bottom">
+                                <field name="display_name"/>
+                                <field name="reference"/>
+                            </tree>
+                       </field>
+                   </form>`,
+            archs: {
+                'partner,false,form': '<form><field name="display_name" widget="test"/></form>',
+            },
+            mockRPC: function (route, args) {
+                if (args.method === 'get_formview_id') {
+                    return Promise.resolve(false);
+                }
+                return this._super(route, args);
+            },
+            res_id: 17,
+        });
+        // current form
+        await testUtils.form.clickEdit(form);
+
+        // open the modal form view of the record pointed by the reference field
+        await testUtils.dom.click(form.$('table td[title="first record"]'));
+        await testUtils.dom.click(form.$('table td button.o_external_button'));
```

Comment:
you don't need a one2Many then... right? Try to simplify the test as much as possible.

Reply:
if I need it because it is the modal which made the error in my case. And for the rest I don't know if that will make my bug. The current test does trigger the bug that I fixed.

Reply:
you open a model from a reference field (basically, you open a many2one record). You don't need the one2many I think.

Reply:
What does not change anything, the test works for the modal. I looked at the test there was at this location. There is little testing and I admit that I don't want to try to reproduce a bug that will not be the one I had. Here I had the bug when opening a modal with a many2many.